modules/network/src/main/java/org/apache/ignite/internal/network/serialization/ClassDescriptor.java

```diff
@@ -52,21 +52,28 @@
     @Nullable
     private final ClassDescriptor superClassDescriptor;
 
+    /**
+     * Component type descriptor (only present for arrays).
+     */
+    @Nullable
+    private final ClassDescriptor componentTypeDescriptor;
+
+    private final boolean isPrimitive;
+    private final boolean isArray;
+    private final boolean isRuntimeEnum;
+    private final boolean isRuntimeTypeKnownUpfront;
+
     /**
      * List of the declared class fields' descriptors.
      */
-    @NotNull
     private final List<FieldDescriptor> fields;
 
     /**
      * How the class is to be serialized.
      */
     private final Serialization serialization;
 
-    /**
-     * Whether the class is final.
-     */
-    private final boolean isFinal;
+    private final List<MergedField> mergedFields;
```

Comment:
And this field too

Reply:
The only thing I can invent as a javadoc here is 'The merged fields', it gives no additional information, just duplication. If something is missing, could you please suggest what you'd like to see in the Javadoc?